File: java-package/brainflow/src/main/java/BoardData.java
import java.util.ArrayList;
import java.util.List;


public class BoardData {

	private List<ArrayList<Double>> board_data = new ArrayList<ArrayList<Double>> ();
	private int package_length;
	
	public BoardData (int package_length, float[] raw_data, double[] ts_data) {
		this.package_length = package_length;
		for (int i = 0; i < ts_data.length; i++) {
			ArrayList<Double> temp = new ArrayList<Double> ();
			for (int j = 0; j < this.package_length; j++) {
				temp.add ((double) raw_data[i * package_length + j]);
			}
			temp.add (ts_data[i]);
			board_data.add (temp);
		}
	}

	@Override
    public String toString() {
		StringBuilder str_b = new StringBuilder ();
		for (ArrayList<Double> row : board_data) {
			for (Double data_point : row) {
				str_b.append (data_point).append (",");
			}
			str_b.append("\n");
		}
        return str_b.toString (); 
    }

	public List<ArrayList<Double>> get_board_data () {
		return board_data;
	}

	public int get_size () {
		return board_data.size ();
	}

	public int get_package_length () {
		return package_length;
	}

}

File: csharp-package/brainflow/brainflow/board_data.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Accord.Math;

namespace brainflow
{
    internal class BoardData
    {
        private double[,] board_data;
        private int package_length;
        private int board_id;

        internal BoardData (int board_id, float[] raw_data, double[] ts_data)
        {
            this.board_id = board_id;
            this.package_length = BoardInfoGetter.get_package_length (board_id);
            List<List<double>> temp_board_data = new List<List<double>> ();
            for (int i = 0; i < ts_data.Length; i++)
            {
                List<double> temp = new List<double> ();
                for (int j = 0; j < this.package_length; j++)
                {
                    temp.Add ((double) raw_data[i * package_length + j]);
                }
                temp.Add (ts_data[i]);
                temp_board_data.Add (temp);
            }
            board_data = temp_board_data.Select (a => a.ToArray ()).ToArray ().ToMatrix ();
        }

        internal double[,] get_board_data ()
        {
            return board_data;
        }
    }
}
